### DISCUSSION TITLE
Is it possible to use Postgres extensions in supabase js

### DISCUSSION THREAD

**Discussion by Praveen-Pandiyan** [supabase/supabase#1540] [Questions] [answered]
like 
 `let { data, error } = await supabase
            .from('search').select('keys').where(supabase.levenshtein('hello') <= 3)
console.log(data)`

**Comment by inian** [comment, marked as answer]
HI Praveen, you can create a stored procedure which uses any function provided by the extension and call it via `supabase.rpc`.

**Reply by Praveen-Pandiyan** [reply]
Thank you @inian 
  but can you please explain more about how to do that ? i am migrated from firebase environment and i had no experience with sql.
thanks in advance

**Reply by inian** [reply]
This is how you create a function. You can create functions in the SQL editor in the dashboard or by connecting to your database directly 
```
create function public.find_distance() 
returns text as $$
begin
  SELECT levenshtein(str1, str2)::text;
end;
$$ language plpgsql security definer;
```

You can call this function from the frontend like ([docs](https://supabase.io/docs/postgrest/client/rpc))
```
await supabase.rpc('find_distance`, { arg: value })
```

**Reply by Praveen-Pandiyan** [reply]
it works...!
thank you 🙏
